src/MultiRegion/multi_region_split_explicit_free_surface.jl

```diff
@@ -12,14 +12,15 @@ import Oceananigans.Models.HydrostaticFreeSurfaceModels: materialize_free_surfac
 function materialize_free_surface(free_surface::SplitExplicitFreeSurface{extend_halos}, 
                                   velocities, 
                                   grid::MultiRegionGrids) where {extend_halos}
+    substepping = free_surface.substepping
 
-    free_surface.substepping isa FixedTimeStepSize &&
+    substepping isa FixedTimeStepSize &&
         throw(ArgumentError("SplitExplicitFreeSurface on MultiRegionGrids only suports FixedSubstepNumber; re-initialize SplitExplicitFreeSurface using substeps kwarg"))
```

Comment:
I don't think this is required anymore, with the `fill_halo_regions!` version we can allow a variable timestep size